### PR TITLE
fix: homepage - use `Link` for internal links

### PR DESCRIPTION
**Pull request by SutuSebastian**
- [x] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

Use NextJS `Link` component for internal routes. Avoid full page refresh and improve UX.

### Before

https://github.com/themesberg/flowbite-react/assets/41998826/f971741b-80f3-4aca-89c0-a1dc1764e95a

### After

https://github.com/themesberg/flowbite-react/assets/41998826/b52b5211-7335-4a58-b7b3-3ebee3a731b3

### CODE REVIEW

**Review by rluders — CHANGES_REQUESTED** [review]

File: app/page.tsx [hunk 1]
```diff
@@ -172,9 +172,11 @@ const HeroSection: FC = () => {
                 </Tooltip>
                 <div className="justify-center sm:flex sm:justify-start">
                   <div className="mx-0 flex flex-row items-center gap-4 sm:gap-6">
-                    <Button href="/docs/getting-started/introduction" size="lg" className="w-full whitespace-nowrap">
-                      Get started <HiOutlineArrowRight className="ml-2 mt-1 h-4 w-4" />
-                    </Button>
+                    <Link href="/docs/getting-started/introduction" tabIndex={-1}>
```
[the remark this answers is not in the document]
Comment (SutuSebastian) [reply app/page.tsx:175]: Already discussed in Discord in the `react-library`, TS throws errors when doing that, it does not properly infer & extend the `Link` component from NextJS. That is the reason I made it like this.

Do u think we should fix the underlying issue with TS first? Luckily this is the only place `href` props is used that is visible to the user.

[the remark this answers is not in the document]
Comment (SutuSebastian) [reply app/page.tsx:175]: Screenshot 2023-09-26 at 19 35 30

[the remark this answers is not in the document]
Comment (rluders) [reply app/page.tsx:175]: As we discussed, I think that we should first fix the button. I'm converting this PR to draft till we get the fix.

[the remark this answers is not in the document]
Comment (nigellima) [reply app/page.tsx:186]: #1244 should fix this now @SutuSebastian @rluders

**Comment by SutuSebastian** [comment]
deprecated by #1246, this one is too outdated.

File: app/page.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (SutuSebastian) [reply app/page.tsx:186]: will push the fixes in #1246.

**Pull request closed without merging**